osquery/core/windows/bstr.h

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright (c) 2014-present, The osquery authors
+ *
+ * This source code is licensed as defined by the LICENSE file found in the
+ * root directory of this source tree.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
+ */
+
+#include <osquery/utils/system/system.h>
+
+#include <string>
+
+#include <wtypes.h>
+
+#include <osquery/utils/only_movable.h>
+
+namespace osquery {
+
+class Bstr : private only_movable {
+ public:
+  static Bstr fromString(std::wstring_view);
+
+  Bstr() = default;
+  ~Bstr();
+
+  explicit Bstr(BSTR bs);
+
+  Bstr(Bstr&& other);
+  Bstr& operator=(Bstr&&);
+
+  explicit operator bool() const {
+    return !!bstr_;
+  }
+
+  BSTR get() const {
```

Comment:
Could you please clarify why `const` should  be added?
`const BSTR` is just a `WCHAR* const` type and in my opinion adding `const` does not affect anything here.
It will be possible to store `const BSTR` result into `BSTR bs` variable, see https://godbolt.org/z/v74cvjbxz

Reply:
> Could you please clarify why const should be added?

I agree that BSTR string can be modified since BSTR is a pointer to a wide-character string, the const qualifier applies to the pointer itself and not the data it points to. The data can still be modified. The const on the return type here is mostly to tell the compiler BSTR that's returned will not be modified (or rather, should not be modified) by the callers. This might not apply here, but when dealing with getters I think it is always good to set the contract upfront and just use const whenever we expect the returned data should not change (return type could be something different in the future and not a pointer)